src/webapi.js

```diff
@@ -40,4 +41,42 @@ router.get('/callback', async (req, res) => {
   res.redirect(`/clips`);
 });
 
+router.get('/refresh', async (req, res) => {
+  if (!req.cookies.discord_session || !req.cookies.discord_session.rt) {
+    return res.status(403).send("Session expired, please log back in");
+  };
+  const refreshToken = req.cookies.discord_session.rt;
+  const form = new URLSearchParams({
+    'client_id': nconf.get('CLIENT_ID'),
+    'client_secret': nconf.get('CLIENT_SECRET'),
+    'grant_type': 'refresh_token',
+    'refresh_token': refreshToken
+  });
+  const response = await fetch(`https://discordapp.com/api/oauth2/token?redirect_uri=${getRedirect(req)}`, {
+    method: 'POST',
+    body: form
+  }).catch(err => {
+    log.debug(`error from discord: ${err}`);
+    res.cookie('discord_session', {}, {
+      maxAge: 9000000, httpOnly: true
+    });
+    return res.status(403).send("Session expired, please log back in");
+  });
+  const json = await response.json();
+  if (!json.error) {
```

Comment:
If there is an error, we should probably send them back to the login page, rather than to /clips or to the callback URL.